terraform/batch-processing/s3.tf

```diff
@@ -0,0 +1,18 @@
+resource "aws_s3_bucket" "batch_lambda_source_bucket" {
+    bucket = "${var.prefix}-batch-lambda-source"
+    force_destroy = true
```

Comment:
This shouldn't be forced if the environment (prefix in this case) is not pr. There is a ternary operator in terraform that you can use. 
The teardown shouldn't happen for any non-pr/shortcode deployment. So, we want to prevent S3 deletion in case it's not empty.